### PR TITLE
conditional execution with the `&&` `||` and `;` operators

### PR DESCRIPTION
**Pull request by sadanslargehole**
implements #8

### CODE REVIEW

**Comment by keli5** [comment]
Breaks `cd ~`

**Comment by sadanslargehole** [comment]
how tf did i manage that

**Comment by sadanslargehole** [comment]
should be fixed

**Comment by keli5** [comment]
- Attempting to cd to a nonexistent subdirectory of `~` like `~/pr` causes a segfault
- Trying to use `cd ~ && ls` seems to cause a segfault, but only under conditions I haven't fully figured out how to reproduce?...

**Comment by keli5** [comment]
```
[letters@letters-manjaro] ~/projects/kcsh
$ cd ~ && ls
ls: cannot access '~': No such file or directory
[letters@letters-manjaro] ~
$ cd ~/projects
[letters@letters-manjaro] ~/projects
$ cd ~ && ls
[1]    203871 segmentation fault (core dumped)  ./build/debug/kcsh-x86_64-debug
```

**Comment by sadanslargehole** [comment]
let me know if you can reproduce

**Comment by sadanslargehole** [comment]
> * Attempting to cd to a nonexistent subdirectory of `~` like `~/pr` causes a segfault


cant reproduce
image

**Comment by keli5** [comment]
both issues mentioned seem to be fixed by 335170e

**Comment by sadanslargehole** [comment]
should be good:tm:

**Review by keli5 — APPROVED** [review]
approved, handle conflicts and then will merge

**Review by keli5 — APPROVED** [review]
approved
